### DISCUSSION TITLE
Details about how tsinfer does imputation

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/tsinfer#619] [Q&A]
@chlangley asked how tsinfer does imputation. This isn't actually documented anywhere, so it would be good to have a documentation section on this, with a few details, e.g. in the "inference overview" section. Any text in the discussion below can be used to fill out the docs.

**Comment by chlangley** [comment]
Found it.  Thanks.

**Comment by hyanwong** [comment]
Incidentally, to replace imputed data with the original missing data, we would need to address https://github.com/tskit-dev/tsinfer/issues/224

**Comment by hyanwong** [comment]
Pinging @szhan as he might start to write a tutorial on this 🎉

**Reply by hyanwong** [reply]
(Sadly there's no material posted here yet, but this GH discussion could serve as a place to talk about it.)

**Comment by mylena-s** [comment]
Dear @hyanwong, 
Is there a way to know what sites were imputed during inference?

I guess I could recover the position of sites with missing data or missing ancestral alleles from the original vcz before inference and generate a mask ... but I was wondering if tsinfer keeps this information anywhere in the ts metadata?

Thanks in advance

**Reply by hyanwong** [reply]
I'm afraid there is no way to find this out easily, other than by examining the original vcz file. There is a problem with storing this information in metadata. Specifically, in the limit of having a large number of sites missing at random among the samples, recording this information may take almost as much space as storing the full genotype matrix (which is explicitly what tskit is designed to avoid).

It could be that, if the missing (imputed) sites concern the same sets of samples, or if the number of samples with missing site data is small, we could record this efficiently. I'm not sure how we could ensure this in the general case, however. If you have any ideas for how this could be done, it could be helpful.

**Reply by hyanwong** [reply]
(NB, the case of having a WGS reference panel and augmenting that with a large number of SNP-array-based samples is a possibly frequently-used case that we could try to figure out, I suppose)

**Reply by mylena-s** [reply]
Thanks for the quick response!

I understand and it totally makes sense not to keep the information in the ts. I will retrieve the info from the original vcz. 

Thanks!
